File: java/test/io/agora/media/DynamicKeyTest.java
package io.agora.media;

import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by liwei on 5/4/16.
 */
public class DynamicKeyTest {

    @Test
    public void testGenerate() throws Exception {
        String vendor = "970ca35de60c44645bbae84215061b33b";
        String key = "5cfd2fd1755d40ecb72977518be15d3b";
        String channel = "7d72365eb983485397e3e3f9d460bdda";
        int ts = 1446455472;
        int r = 58964981;
        int uid = 999;
        int expiredTs = 1446455471;
        String result = DynamicKey.generate(vendor, key, channel, ts, r);

        String expected = "70e0e5b0a137630190bfa475506de9fb56012576970ca35de60c44645bbae84215061b33b14464554720383bbf5";
        assertEquals(expected, result);
    }
}

File: nodejs/test/DynamicKeyTest.js
/**
 * run this test with command:
 * nodeunit DynamicKeyTest.js
 * see https://github.com/caolan/nodeunit
 */
var DynamicKey = require('../src/DynamicKey');
var DynamicKey3 = require('../src/DynamicKey3');
var DynamicKey4 = require('../src/DynamicKey4');
var vendor  = "970ca35de60c44645bbae8a215061b33";
var key     = "5cfd2fd1755d40ecb72977518be15d3b";
var channel = "7d72365eb983485397e3e3f9d460bdda";
var ts = 1446455472;
var r = 58964981;
//var uid=999;
var uid=2882341273;
var expiredTs=1446455471;

exports.DynamicKeyTest = function (test) {
  var expected = "870588aad271ff47094eb622617e89d6b5b5a615970ca35de60c44645bbae8a215061b3314464554720383bbf5";
  var actual = DynamicKey.generate(vendor, key, channel, ts, r);
  test.equal(expected, actual);
  test.done();
};


exports.DynamicKey3Test = function (test) {
  var expected = "0037666966591a93ee5a3f712e22633f31f0cbc8f13970ca35de60c44645bbae8a215061b3314464554720383bbf528823412731446455471";
  var actual = DynamicKey3.generate(vendor, key, channel, ts, r, uid, expiredTs);
  test.equal(expected, actual);
  test.done();
};


exports.RecordingKeyTest = function (test) {
  var expected = "004e0c24ac56aae05229a6d9389860a1a0e25e56da8970ca35de60c44645bbae8a215061b3314464554720383bbf51446455471";
  var actual = DynamicKey4.generateRecordingKey(vendor, key, channel, ts, r, uid, expiredTs);
  test.equal(expected, actual);
  test.done();
};

exports.MediaChannelKeyTest = function (test) {
  var expected = "004d0ec5ee3179c964fe7c0485c045541de6bff332b970ca35de60c44645bbae8a215061b3314464554720383bbf51446455471";
  var actual = DynamicKey4.generateMediaChannelKey(vendor, key, channel, ts, r, uid, expiredTs);
  test.equal(expected, actual);
  test.done();
};
